chasm/tree.go

```diff
@@ -155,12 +164,88 @@ func NewEmptyTree(
 
 // Component retrieves a component from the tree rooted at node n
 // using the provided component reference
-// It also performs consistency, access rule, and task validation checks
+// It also performs access rule, and task validation checks
 // (for task processing requests) before returning the component.
 func (n *Node) Component(
 	chasmContext Context,
 	ref ComponentRef,
 ) (Component, error) {
+	node, ok := n.getNodeByPath(ref.componentPath)
+	if !ok {
+		return nil, errComponentNotFound
+	}
+
+	metadata := node.persistence.Metadata
+	if ref.componentInitialVT != nil && transitionhistory.Compare(
+		ref.componentInitialVT,
+		metadata.InitialVersionedTransition,
+	) != 0 {
+		return nil, errComponentNotFound
+	}
+
+	value, err := node.prepareComponentValue(chasmContext)
+	if err != nil {
+		return nil, err
+	}
+
+	componentValue, ok := value.(Component)
+	if !ok {
+		return nil, serviceerror.NewInternal(
+			fmt.Sprintf("component value is not of type Component: %v", reflect.TypeOf(node.value)),
+		)
+	}
+
+	// TODO: perform access rule check based on the operation intent
+	// and lifecycle state of all ancenstor nodes.
+	//
+	// intent := operationIntentFromContext(chasmContext.getContext())
+	// if intent != OperationIntentUnspecified {
+	// 	...
+	// }
+
+	if ref.validationFn != nil {
+		if err := ref.validationFn(chasmContext, componentValue); err != nil {
+			return nil, err
+		}
+	}
+
+	return componentValue, nil
+}
+
+func (n *Node) prepareComponentValue(
```

Comment:
This method or at least some part of it can be reused by Field.Get() implementation I think. cc @alexshtin